subxt/src/blocks/extrinsic_types.rs

```diff
@@ -558,6 +607,117 @@ impl<T: Config> ExtrinsicEvents<T> {
+        let signed_extension_types = self.metadata.extrinsic().signed_extensions();
+        let num_signed_extensions = signed_extension_types.len();
+        let mut index = 0;
+        let mut byte_start_idx = 0;
+
+        std::iter::from_fn(move || {
+            if index == num_signed_extensions {
+                None
+            } else {
+                let extension = &signed_extension_types[index];
+                let ty_id = extension.extra_ty();
+                let cursor = &mut &self.bytes[byte_start_idx..];
+                if let Err(err) = scale_decode::visitor::decode_with_visitor(
+                    cursor,
+                    ty_id,
+                    self.metadata.types(),
+                    scale_decode::visitor::IgnoreVisitor,
+                )
+                .map_err(|e| Error::Decode(e.into()))
+                {
+                    index = num_signed_extensions; // (such that None is returned in next iteration)
+                    return Some(Err(err));
+                }
+                let byte_end_idx = self.bytes.len() - cursor.len();
+                let bytes = &self.bytes[byte_start_idx..byte_end_idx];
+                byte_start_idx = byte_end_idx;
+                index += 1;
+                Some(Ok(ExtrinsicSignedExtension {
+                    bytes,
+                    ty_id,
+                    identifier: extension.identifier(),
+                    metadata: self.metadata.clone(),
+                }))
+            }
+        })
+    }
+
+    /// The tip of an extrinsic, extracted from the ChargeTransactionPayment signed extension.
+    pub fn tip(&self) -> Option<u128> {
+        let tip = self
+            .find("ChargeTransactionPayment")
+            .or_else(|| self.find("ChargeAssetTxPayment"))?;
+        // Note: ChargeAssetTxPayment might have addition information in it (asset_id).
+        //       But both should start with a compact encoded u128, so this decoding is fine.
+        let tip = Compact::<u128>::decode(&mut tip.bytes()).ok()?.0;
+        Some(tip)
+    }
+
+    /// The nonce of the account that submitted the extrinsic, extracted from the CheckNonce signed extension.
+    pub fn nonce(&self) -> Option<u64> {
+        let nonce = self.find("CheckNonce")?;
+        let nonce = Compact::<u64>::decode(&mut nonce.bytes()).ok()?.0;
+        Some(nonce)
+    }
+}
+
+impl<'a> ExtrinsicSignedExtension<'a> {
+    /// The extra bytes associated with the signed extension.
+    pub fn bytes(&self) -> &[u8] {
+        self.bytes
+    }
+
+    /// The name of the signed extension.
+    pub fn name(&self) -> &str {
+        self.identifier
+    }
+
+    /// The type id of the signed extension.
+    pub fn type_id(&self) -> u32 {
+        self.ty_id
+    }
+
+    /// DecodedValueThunk representing the type of the extra of this signed extension.
+    pub fn decoded(&self) -> Result<DecodedValueThunk, Error> {
```

Comment:
Let's remove thid `decoded` fn; it doesn't add much on top of `value()` below imo :)